jsonschema/src/compilation/options.rs

```diff
@@ -79,6 +102,11 @@ impl CompilationOptions {
         let resolver = Resolver::new(draft, &scope, schema, self.store.clone())?;
         let context = CompilationContext::new(scope, processed_config);
 
+        META_SCHEMA_VALIDATORS
+            .get(&draft)
+            .expect("existing draft!")
+            .validate(schema)
+            .expect("valid meta-schema!");
```

Comment:
@Stranger6667 Running `cargo check` outputs this:
```
PS C:\rust\jsonschema-rs\jsonschema> cargo check
    Checking jsonschema v0.9.0 (C:\rust\jsonschema-rs\jsonschema)
error[E0599]: the method `expect` exists for enum `Result<(), Box<dyn Iterator<Item = ValidationError<'_>> + Send + Sync>>`, but its trait bounds were not satisfied
   --> src\compilation\options.rs:109:14
    |
109 |               .expect("valid meta-schema!");
    |                ^^^^^^ method cannot be called on `Result<(), Box<dyn Iterator<Item = ValidationError<'_>> + Send + Sync>>` due to unsatisfied trait bounds
    | 
   ::: C:\Users\edeny\.rustup\toolchains\stable-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\alloc\src\boxed.rs:165:1
    |
165 | / pub struct Box<
166 | |     T: ?Sized,
167 | |     #[unstable(feature = "allocator_api", issue = "32838")] A: Allocator = Global,
168 | | >(Unique<T>, A);
    | |________________- doesn't satisfy `_: Debug`
    |
    = note: the following trait bounds were not satisfied:
            `Box<dyn Iterator<Item = ValidationError<'_>> + Send + Sync>: Debug`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0599`.
error: could not compile `jsonschema`

To learn more, run the command again with --verbose.
```

I will continue investigating it but I have a good feeling it's related to something small that I'm not familiar with in Rust yet. Please let me know if you have any suggestions regarding this.

Reply:
I think it could be something like this:

```rust
if let Some(mut errors) = META_SCHEMA_VALIDATORS
    .get(&draft)
    .expect("existing draft!")
    .validate(schema)
    .err()
{
    return Err(errors.next().expect("Should have at least one element"));
}
```

And it will require changing the return type a bit - `Result<JSONSchema<'a>, ValidationError<'a>>`. However, it leads to an error in `main.rs` that I don't know how to solve yet.

Reply:
In the `main.rs` file the problem could be solved like this:

```rust
fn validate_instances(instances: &[PathBuf], schema: PathBuf) -> BoxErrorResult<bool> {
    let mut success = true;

    let schema_json = fs::read_to_string(schema)?;
    let schema_json = serde_json::from_str(&schema_json)?;
    match JSONSchema::compile(&schema_json) {
        Ok(schema) => {
            for instance in instances {
                let instance_path_name = instance.to_str().unwrap();
                let instance_json = fs::read_to_string(&instance)?;
                let instance_json = serde_json::from_str(&instance_json)?;
                let validation = schema.validate(&instance_json);
                match validation {
                    Ok(_) => println!("{} - VALID", instance_path_name),
                    Err(errors) => {
                        success = false;

                        println!("{} - INVALID. Errors:", instance_path_name);
                        for (i, e) in errors.enumerate() {
                            println!("{}. {}", i + 1, e);
                        }
                    }
                }
            }
        }
        Err(error) => {
            println!("Schema is invalid. Error: {}", error);
            success = false
        }
    }
    Ok(success)
}
```

I.e. we match on the compilation result and in case of error print that the schema is not valid and return false to avoid lifetime issues with returning a reference to the value owned by that function. Otherwise, we run the old code in the `Ok` branch.